.github/workflows/main.yaml

```diff
@@ -0,0 +1,42 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - '*'
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  # Go needs absolute directories, using the $HOME variable doesn't work here.
+  GOPATH: /home/runner/work/go
+
+  GO_VERSION: '1.22.11'
```

Comment:
Better to get it from go.mod? https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file